.changeset/mighty-squids-count.md

```diff
@@ -0,0 +1,15 @@
+---
+"@cloudflare/sandbox": patch
+---
+Add process readiness detection with port and log pattern waiting
+The `Process` object returned by `startProcess()` now includes readiness methods:
+
+  - `process.waitForPort(port, options?)`: Wait for process to listen on a port
+    - Supports HTTP mode (default): checks endpoint returns expected status (200-399)
+    - Supports TCP mode: checks port accepts connections
+    - Container-side checking via `/api/port-check` endpoint
```

Comment:
nit: This is an internal implementation, we can remove from changeset to avoid confusions for users.